applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_element.cpp

```diff
@@ -1580,6 +1599,30 @@ Vector UPwSmallStrainElement<TDim, TNumNodes>::CalculateGreenLagrangeStrain(cons
     return this->GetStressStatePolicy().CalculateGreenLagrangeStrain(rDeformationGradient);
 }
 
+template <unsigned int TDim, unsigned int TNumNodes>
+std::vector<double> UPwSmallStrainElement<TDim, TNumNodes>::CalculateDeterminantsOfDeformationGradients(
+    const std::vector<Matrix>& rDeformationGradients) const
+{
+    std::vector<double> result(rDeformationGradients.size());
+    std::transform(
+        rDeformationGradients.cbegin(), rDeformationGradients.cend(), result.begin(),
+        [](const auto& rDeformationGradient) { return MathUtils<>::Det(rDeformationGradient); });
+
+    return result;
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+std::vector<Matrix> UPwSmallStrainElement<TDim, TNumNodes>::CalculateDeformationGradients() const
+{
+    std::vector<Matrix> result;
+    for (unsigned int GPoint = 0;
+         GPoint < this->GetGeometry().IntegrationPointsNumber(this->GetIntegrationMethod()); ++GPoint) {
+        result.push_back(CalculateDeformationGradient(GPoint));
+    }
+
+    return result;
+}
+
```

Comment:
I think this function can be moved to utilities next time. It looks that now functions called by this function become very similar for different elements.